src/components/nav/Footer.js

```diff
@@ -13,27 +13,18 @@ class Footer extends React.Component {
     };
   }
 
-  // componentWillMount(){
-  //   if(this.props.authenticated){
-  //     this.props.getUserName();
-  //   }
-  // }
-
   componentWillReceiveProps(nextProps){
     if(nextProps.currentProject === 0){
       this.setState({isOwner: false});
     }
     if(nextProps.currentProject){
       this.isProjectOwner(nextProps.currentProject);
     }
-    // if(nextProps.authenticated && nextProps.authenticated !== this.props.authenticated){
-    //   this.props.getUserName();
-    // }
   }
 
   isProjectOwner(id){
     let project;
-    let array = this.props.projects;
+    const array = this.props.projects;
     for(let i = 0; i < array.length; i++){
```

Comment:
I would like to challenge you to try using [].find here.

Reply:
On line 35, try getting rid of the `if` statement and do a single `this.setState()` with the conditional as the value for `isOwner`.